services/common/src/main/java/org/finra/gatekeeper/common/services/user/auth/GatekeeperAuthorizationService.java

```diff
@@ -75,7 +77,11 @@ public Optional<Set<String>> load(String userName) throws Exception {
     }
 
     public Set<String> getMemberships(){
-        return userMembershipCache.getUnchecked(gatekeeperUserProfileSupplier.get().getName()).get();
+        return useHeadersForMembership ? getMembershipsFromHeader() : userMembershipCache.getUnchecked(gatekeeperUserProfileSupplier.get().getName()).get();
```

Comment:
This kinda goes against the pattern that I had set in place here. The getMembershipsFromHeader() method should be implemented within a class that extends GatekeeperAuthorizationService